### PR TITLE
Presnet host detials on info page

### PR DESCRIPTION
**Pull request by olivergondza**
In order to implement that, internal representation of a host had to change.

- `HostInfo` is now a snapshot of foreman host details preserved at Jenkins side.
  - All queries for individual attributes ware replaced by this one.
- `Cloud#hostsMap` represents possibly outdated hosts *inventory*. It is updated often enough to pick changes to the inventory itself (what nodes are there, their attributes) but the reservation status needs to be re-queried.

- Test updates
  - Use random wire mock ports so tests can be run in parallel
  - Support wire mock configuration from multiple threads - namely Jenkins build
  - Avoid large shared fixture
    - Rename files so it is clear what responses they represent
    - Remove replies no longer needed by the plugin
    - Provide fine grained stub methods to configure foreman state as desired on per-test basis
    - Add reusable scenarios for reservation/release

I wanted this to be a small change but I failed miserably.  Do not open the can of worms they said...

### CODE REVIEW

**Comment by olivergondza** [comment]
@pjanouse, @scoheb, please have a look. Thanks.

File: src/test/java/com/redhat/foreman/ForemanSharedNodeCloudTest.java [hunk 1]
```diff
@@ -82,112 +104,42 @@
...
-                        .withHeader("Content-Type", "text/json")
-                        .withBody(body1)));
-
-        stubFor(get(urlEqualTo("/api/v2/hosts?search=params.JENKINS_LABEL%3Dlabel1+and+params.RESERVED%3D"
-                + "false+and+has+params.JENKINS_SLAVE_REMOTEFS_ROOT"))
-                .willReturn(aResponse()
-                        .withStatus(HTTPOK)
-                        .withHeader("Content-Type", "text/json")
-                        .withBody(body2)));
+                .willReturn(ok("body1.txt")));
```
Comment (scoheb) [inline src/test/java/com/redhat/foreman/ForemanSharedNodeCloudTest.java]: I am not clear if "bodyX.txt" is valid here? Seems like they got renamed?

**Comment by scoheb** [comment]
Nice changes! LGTM

File: src/test/java/com/redhat/foreman/ForemanSharedNodeCloudTest.java (hunk 1 above)
Comment (olivergondza) [inline src/test/java/com/redhat/foreman/ForemanSharedNodeCloudTest.java]: I have only named those I needed. The method and unrenamed resources are hereby unused and kept not to throw them away. I expect they will be renamed and reused as new tests are written.

**Pull request merged**